README.md

```diff
@@ -48,11 +48,45 @@ secret key used to authenticate the session. Inside the handler, we call
 session values in session.Values, which is a `map[interface{}]interface{}`.
 And finally we call `session.Save()` to save the session in the response.
 
-Important Note: If you aren't using gorilla/mux, you need to wrap your handlers
-with
+## Avoiding Memory Leaks
+
+If you are not using gorilla/mux (or are using context), you will
+need to perform additional cleanup or else you will leak memory!
+
+Here is an example that illustrates proper cleanup:
+
+```go
+	import (
+		"context"
+		"net/http"
+
+		gorillacontext "github.com/gorilla/context"
+		"github.com/gorilla/sessions"
+	)
+
+	var store = sessions.NewCookieStore([]byte("something-very-secret"))
+
+	func MyHandler(w http.ResponseWriter, r *http.Request) {
+		// Store some context on the request. This creates
+		// a shallow copy of the request object.
+		r = r.WithContext(context.TODO())
+
+		// Get a session. We're ignoring the error resulted from decoding an
+		// existing session: Get() always returns a session, even if empty.
+		session, _ := store.Get(r, "session-name")
+
+		// Clear the registry at the end of the request lifecycle
+		defer gorillacontext.Clear(r)
```

Comment:
Note that there's two things here:

* gorilla/context
* the modern Go 1.7 request.context, which uses pkg context

Both do NOT co-exist at the same time within mux: if you are using >= Go 1.7, mux will always use the modern context. Leaks ONLY arise if you have other packages using the gorilla/context map in conjunction with a package that uses the context package - that results in the gorilla context map being "islanded" on the old request (pre-copy) and thus may cause leaks.

What other packages are you using?